### DISCUSSION TITLE
Example of creating basic auth credentials

### DISCUSSION THREAD

**Discussion by freshollie** [static-web-server/static-web-server#106] [Q&A] [answered]
I assume:

1. `htpasswd -nbBC 10 username password`

Outputs `username:hashedpassword`

~~2. `echo -n "username" | base64 && echo -n "hasedpassword" | base64`~~
~~3. Set basic auth as `base64username:base64password`?~~

The docs don't make the process clear (~~especially around base64 encoding~~) would be good to clarify?

**Comment by freshollie** [comment]
FYI, when I try the above authentication doesn't work with the username and password, and if I base64 encode both the username and password together it doesn't like the formatting and throws an error.

**Comment by freshollie** [comment, marked as answer]
Ah, did the base64 part mentioned in the docs just mean on the request side?

**Reply by joseluisq** [reply]
I did get it at all, what do you mean exactly with "on the request side"?

**Reply by freshollie** [reply]
https://sws.joseluisq.net/configuration/environment-variables/#server_basic_auth

> It provides [The "Basic" HTTP Authentication Scheme](https://datatracker.ietf.org/doc/html/rfc7617) using credentials as user-id:password pairs, encoded using Base64

It wasn't clear if the environment variable must be base64 encoded. It turns out it shouldn't.

**Reply by joseluisq** [reply]
Ah that one. Yes you are right. I think that part is confusing since as you said, base64 is only required when clients request something to sws.

Happy to see some PR clarifying that.

**Comment by joseluisq** [comment]
BTW if you found some issues on the docs just feel free to send a PR to improve it.

**Reply by freshollie** [reply]
Would be good to get clarity on what you intended in the docs, then I'll submit an improvement :)

**Reply by joseluisq** [reply]
Great, go ahead!

**Reply by joseluisq** [reply]
Maybe it deserves to give a check to the whole basic auth page.

**Reply by joseluisq** [reply]
https://sws.joseluisq.net/building-from-source/#development-server

Use that as reference to improve the docs.
